homework/inner-product/arithmeticAverage.cpp

```diff
@@ -0,0 +1,26 @@
+#include "arithmeticAverage.hpp"
+
+#include <cmath>
+#include <functional>
+#include <numeric>
+#include <stdexcept>
+
+double ArithmeticAverage(const std::vector<int> &first, const std::vector<int> &second) {
+  double sum = std::reduce(first.cbegin(), first.cend(), 0) +
+               std::reduce(second.cbegin(), second.cend(), 0);
+  return sum / (first.size() + second.size());
+}
+
+double Distance(const std::vector<int> &first, const std::vector<int> &second) {
+  if (first.size() != second.size()) {
+    throw std::logic_error{"vectors' sizes differ"};
+  }
+  return std::sqrt(std::transform_reduce(first.cbegin(), 

```

Comment:
Nice one :) but we should you inner product as in homework description. But this is also a good sollution